compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -11824,39 +11827,9 @@ OMR::Z::TreeEvaluator::primitiveArraycopyEvaluator(TR::Node* node, TR::CodeGener
    }
 
 void
-OMR::Z::TreeEvaluator::referenceArraycopyEvaluator(TR::Node* node, TR::CodeGenerator* cg, TR::Node* byteSrcNode, TR::Node* byteDstNode, TR::Node* byteLenNode, TR::Node* byteSrcObjNode, TR::Node* byteDstObjNode)
+OMR::Z::TreeEvaluator::referenceArraycopyEvaluator(TR::Node* node, TR::CodeGenerator* cg)
    {
-   TR::Register* byteSrcObjReg = cg->evaluate(byteSrcObjNode);
-   TR::Register* byteDstObjReg = cg->evaluate(byteDstObjNode);
-
-   // For reference array copies that need an array store check always call the helper function
-   if (!node->isNoArrayStoreCheckArrayCopy())
-      {
-      TR::Register* byteSrcReg = cg->evaluate(byteSrcNode);
-      TR::Register* byteDstReg = cg->evaluate(byteDstNode);
-      TR::Register* byteLenReg = cg->evaluate(byteLenNode);
-
-#ifdef J9_PROJECT_SPECIFIC
-      TR::TreeEvaluator::genArrayCopyWithArrayStoreCHK(node, byteSrcObjReg, byteDstObjReg, byteSrcReg, byteDstReg, byteLenReg, cg);
-#endif
-
-      cg->decReferenceCount(byteSrcNode);
-      cg->decReferenceCount(byteDstNode);
-      cg->decReferenceCount(byteLenNode);
-      }
-   else
-      {
-      TR_ASSERT_FATAL(node->getArrayCopyElementType() == TR::Address, "Reference arraycopy element type should be TR::Address but was '%s'", node->getArrayCopyElementType().toString());
-
-      TR::TreeEvaluator::primitiveArraycopyEvaluator(node, cg, byteSrcNode, byteDstNode, byteLenNode);
-
-#ifdef J9_PROJECT_SPECIFIC
-      TR::TreeEvaluator::genWrtbarForArrayCopy(node, byteSrcObjReg, byteDstObjReg, byteSrcNode->isNonNull(), cg);
-#endif
-      }
-
-   cg->decReferenceCount(byteSrcObjNode);
-   cg->decReferenceCount(byteDstObjNode);
+   TR_UNIMPLEMENTED();
```

Comment:
Given our offline chat it sounded like reference arraycopy does not belong in OMR at all. We should just remove it altogether and override `arraycopyEvaluator` in OpenJ9. In OpenJ9 we can make the `node->isReferenceArrayCopy()` checks, etc.

Reply:
`referenceArrayCopy` is removed in 309a44a182123a3cee305f32616f5a3b107ce541